clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp

```diff
@@ -1022,9 +1023,13 @@ Expected<StringRef> wrapSYCLBinariesFromFile(StringRef InputFile,
   // spir64-unknown-unknown.
   // TODO: Fix SYCL runtime to accept both triple
   llvm::Triple T(Target);
-  StringRef A(T.getArchName());
+  std::string EmbeddedIRTarget("llvm_");
+  EmbeddedIRTarget.append(T.getArchName());
+  std::string RegularTarget(T.getArchName());
+
```

Comment:
Just a question, does the the Target name we used for the embedded image matter? Does the SYCL runtime ever use the embedded IR somehow?

Reply:
Story behind this. Initially I did not have this change. Several tests (run in AMD backend) failed with kernel not found errors. I checked the old offloading model for any missing code. I saw that we were renaming the embedded image like this. I added that change and the error disappeared. So, it does look like the SYCL runtime does look for these IRs (specifically  in the KernelFusion tests).
Sorry that this is a bit non-technical. I think someone from SYCL runtime team might have a better answer.

Thanks

Reply:
Got it, thanks